Assets/Scripts/Pathfinding/Pathfinder.cs

```diff
@@ -0,0 +1,106 @@
+using System.Collections.Generic;
+using Examen.Pathfinding.Grid;
+using FishNet.Object;
+using UnityEngine;
+
+namespace Examen.Pathfinding
+{
+    public class Pathfinder : NetworkBehaviour
+    {
+        [SerializeField] private int _diagonalCost = 14;
+        [SerializeField] private int _straightCost = 10;
+
+        private GridSystem _gridSystem;
+
+        private void OnEnable() => _gridSystem = FindObjectOfType<GridSystem>();
+
+        /// <summary>
+        /// Finds a path from the specified start position to the target position using the A* pathfinding algorithm.
+        /// </summary>
+        /// <param name="startPos">The starting position.</param>
+        /// <param name="targetPos">The target position.</param>
+        /// <returns>A list of nodes representing the path from the start position to the target position.</returns>
+        [Server]
+        public List<Node> FindPath(Vector3 startPos, Vector3 targetPos)
+        {
+            Node startNode = _gridSystem.GetNodeFromWorldPosition(startPos);
+            Node targetNode = _gridSystem.GetNodeFromWorldPosition(targetPos);
+
+            List<Node> openSet = new();
+            HashSet<Node> closedSet = new();
```

Comment:
Move these to variables and clear them when this function gets called. This generates less garbage